R/generate.R

```diff
@@ -0,0 +1,55 @@
+int_types <- c("integer", "smallint")
+logi_types <- c("boolean")
+num_types <- c("date", "numeric", "real")
+date_types <- c("timestamp without time zone", "timestamp with time zone")
+
+generate_type_check <- function(column_name, data_type, is_nullable) {
+  r_type <- ifelse(data_type %in% int_types, "\"integer\"",
```

Comment:
`ifelse` is evil and surprising (it does awful things with replication and type conversion) and is on my list of "please avoid" functions. You can use `a <- if (cond) b else c` and this is fine.  However, here you could just use an ordinary if/else ladder.  Martin also avoided these because he felt like defining a variable within if/else blocks was ugly, but it is pretty standard even if invalid in any typed language